### PR TITLE
Fix a bug when using CoordinateFrame directly

### PR DESCRIPTION
**Pull request by Cadair**
I don't see any reason why CoordinateFrame should apply axes_order, I believe that's a global thing not per-frame?

### CODE REVIEW

**Comment by chris-simpson** [comment]
There are inconsistencies in the way all this operates #269

**Comment by Cadair** [comment]
Looks like you ran into this bug in your comment in #269. I am not really qualified to answer the more fundamental questions in #269 but at least this chips away one small part of it.

**Comment by Cadair** [comment]
This particular fix got the code to consistency with the other frames (spectral and celestial etc) in the easiest way I could see. I agree it is generally unclear though.

**Comment by Cadair** [comment]
@chris-simpson I have looked over this again today, and I think that your points in #269 are correct and should be implemented. That being said, I consider this a bugfix to the existing behaviour, one that is breaking my code!

If @nden wants to weigh in on #269 I probably have a small amount of time to look at taking things in a different direction, but either way I would like to see this PR land.

File: gwcs/coordinate_frames.py [hunk 1]
```diff
@@ -279,6 +278,10 @@ def coordinate_to_quantity(self, *coords):
         Given a rich coordinate object return an astropy quantity object.
         """
         # NoOp leaves it to the model to handle
+        # If coords is a 1-tuple of quantity then return the element of the tuple
+        # This aligns the behavior with the other implementations
+        if not hasattr(coords, 'unit') and len(coords) == 1:
+            return coords[0]
```
Comment (nden) [inline gwcs/coordinate_frames.py:284]: I'm trying to understand where this is a problem. Within gwcs there are two calls to `coordinate_to_quantity` and both pass  `*coords` . Are you calling it outside gwcs?

Also the new test does not actually test this use case.

Comment (nden) [reply gwcs/coordinate_frames.py:284]: If you can change the test I htin kit's good to go

**Comment by nden** [comment]
I think this fixes a real bug regardless of the general discussion in #269.

File: gwcs/coordinate_frames.py (hunk 1 above)
Comment (Cadair) [reply gwcs/coordinate_frames.py:284]: I have added a test exercising this. I also reproduced the scenario in which I was hitting it just to be safe :rofl:

**Review by nden — APPROVED** [review]

**Pull request merged**